cmd/client/noderule.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package client
+
+import (
+	"os"
+
+	"github.com/skydive-project/skydive/api/client"
+	api "github.com/skydive-project/skydive/api/types"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/topology/enhancers"
+	"github.com/skydive-project/skydive/topology/graph"
+	"github.com/skydive-project/skydive/validator"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	nodeName string
+	nodeType string
+	metadata string
+	query    string
+	action   string
+)
+
+// NodeRuleCmd skydive node rule root command
+var NodeRuleCmd = &cobra.Command{
+	Use:          "node-rule",
+	Short:        "node-rule",
+	Long:         "node-rule",
+	SilenceUsage: false,
+}
+
+// NodeRuleCreate skydive node create command
+var NodeRuleCreate = &cobra.Command{
+	Use:          "create",
+	Short:        "create",
+	Long:         "create",
+	SilenceUsage: false,
+
+	Run: func(cmd *cobra.Command, args []string) {
+		client, err := client.NewCrudClientFromConfig(&AuthenticationOpts)
+		if err != nil {
+			logging.GetLogger().Error(err.Error())
+			os.Exit(1)
+		}
+
+		m, err := usertopology.DefToMetadata(metadata, graph.Metadata{"Name": nodeName})
```

Comment:
1/ DefToMetadata is only used by in the "cmd/client" package so I suggest moving it to it, so you don't have to export it.
2/ Graph.Metadata{} should be empty here, as you are giving the node the name of the rule